0000-account-name-system.md

```diff
@@ -44,11 +44,16 @@ Here, important to note that `CreateAccount` is created by some other account.
 
 The next rules apply for relationship between `originator_id` and `new_account_id`:
 
-- `new_account_id` can only contain characters from the [a-z0-9@._-] set. And be 5-32 characters long.
-- If `new_account_id` is suppose to be TLA, it can not contain any separators (e.g. `@._-`).
-- If `new_account_id` is not suppose to be TLA, it's suffix must contain `'.' + originator_id`. For example, if `originator_id` is `'com'`, then this account can register `google.com` or `facebook.com` but can not register `googlecom` or `facebook@com`.
+- `new_account_id` can only contain characters from the [a-z0-9.@_-] set. And be 2-32 characters long. Where if the length is less than 5, `new_account_id` can not contain '.' separator (e.g. can only be TLA).
+- If `new_account_id` is suppose to be a TLA, it can not contain any separators (e.g. `@._-`).
+- If `new_account_id` is not suppose to be a TLA, it's suffix must contain `'.' + originator_id` and the prefix can not contain '.'s. For example, if `originator_id` is `'com'`:
+    - can register `google.com` or `facebook.com` 
+    - can not register `googlecom` or `facebook@com`.
+    - can not directly register `abc.google.com`.
 
-Because TLAs are important as backbone of naming system, we require that `CreateAccount` transaction that creates TLA domain attach an extra `X` NEAR tokens which will be burned by the system. Where `X` is the function of length of the TLA name.
+Generally the shorter name the more desirable it is. To prevent squatting of this names without using it provide crucial services. To make squatting an expensive procedure, we define additional storage price for account names:
+- if `account_id.length > 10`, storage price is equal to regular byte storage price based on the length of `account_id`.
+- otherwise, rounded down `1,000 / (3 ^ (account_id.length - 2))` of NEAR tokens per year.
```

Comment:
Is this the per byte price?

Reply:
no, this is just an extra storage rent for this account.